module2/lessons/model_testing.md

```diff
@@ -0,0 +1,313 @@
+
+In your `Gemfile` Inside the existing `group :development, :test` block, add
+
+- `gem "rspec-rails"`
+- `gem "pry"`
+- `gem "simplecov"`
+
+Your Gemfile should now have this:
+
+**Gemfile**
+
+```bash
+group :development, :test do
+  # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
+  gem "debug", platforms: %i[ mri mingw x64_mingw ]
+  gem "pry"
+  gem "rspec-rails"
+  gem "simplecov"
+end
+```
+
+Always run `bundle install` whenever you update your Gemfile. Now from the command line run:
+
+```bash
+$ rails g rspec:install
+```
+
+What new files did this generate?
+
+- `./.rspec` file
+- a whole `./spec/` directory
+- `./spec/rails_helper.rb` is the new `spec_helper`, holds Rails-specific configurations
+- `./spec/spec_helper.rb` - where we keep all specs that don't depend on rails
+
+At the top of your `rails_helper.rb`, add these lines to configure SimpleCov:
+
+**spec/rails_helper.rb**
+
+```ruby
+require "simplecov"
+SimpleCov.start
+```
+
+We'll also add a line for `coverage` to the `.gitignore` file so that our SimpleCov reports aren't pushed to GitHub.
+
+## Testing the Task Model
+
+Let's start by creating a test file for the Task class. From the command line run:
+
+```bash
+$ mkdir spec/models
+$ touch spec/models/task_spec.rb
+```
+
+It is important that these folders are named `spec` and `models`, respectively.
+
+In our new test file, add the following:
+
+**spec/models/task_spec.rb**
+
+```ruby
+require "rails_helper"
+
+RSpec.describe Task, type: :model do
+
+end
+```
+
+This is the basic set up for any model test. `RSpec.describe Task` tells our test that we are testing the Task class. `type: :model` tells our test that it is a model test. You can optionally leave this out since our test will recognize that it is a model test because it is defined inside `spec/models`. Yes, the name of the folders and files will affect how your tests run.
+
+### What to Test?
+
+We've seen request specs handle the end-to-end functionality of our API endpoints, so what should we be testing in the model? 
+
+The answer: Data logic and integrity! 
+
+Model tests are where we can isolate data manipulations and make sure they're working correctly. Our request specs are integration tests, while our model tests are more like unit tests. We've seen unit tests before, but now they're just used in the context of a Rails application. 
+
+Here are some common tests you'll find in the model:
+
+* Testing relationships
+* Testing validations
+* Testing instance and class methods
+
+In our [ActiveRecord Associations](https://curriculum.turing.edu/module2/lessons/one_to_many_relationships_part2) class we discuss testing relationships, and we'll get to testing validations in [Data Validations](https://curriculum.turing.edu/module2/lessons/data_validations), so the focus of this lesson plan will be testing instance methods. The good news: we've seen this type of test before in Module 1!
```

Comment:
Do we want absolute or relative links? I thought we were doing relative, but IDK!